### PR TITLE
more things to do with packaging

### PR DESCRIPTION
**Pull request by liam-o-marsh**
- new version of metatensor
- fixed the discrepancies between pip's files, requirements.txt and environment.yml
- added details to readme

### CODE REVIEW

File: qstack/regression/kernel_utils.py [hunk 1]
```diff
@@ -205,7 +205,7 @@ def cdist(X, Y):
       x = np.array([x] * len(Y))
       d = np.abs(x-Y)
       while len(d.shape)>1:
-        d = np.sum(d, axis=1) # several axis available for np > 1.7.0
+        d = np.sum(d, axis=1) # several axis available for np > 1.7.0  (TODO shall we move this)
```
Comment (briling) [inline qstack/regression/kernel_utils.py:208]: yes! I forgot to update

**Review by briling — APPROVED** [review]

**Comment by briling** [comment]
is there a way to auto sync `requirements.txt`, `environment.yml`, and `pyproject.toml`?

**Comment by liam-o-marsh** [comment]
not to my knowledge?
environment.yml and requirements.txt should be somewhat feasible because the idea is fundamentally the same, but syncing this with pyproject.toml sounds more difficult.
At the very least, there should be a way to check that the package list is the same across all three, and that the bounds of the dependencies listed in pyproject.toml are compatible with the rest.

**Comment by briling** [comment]
is it possible to import `qstack_qml` in the main `__init__.py` to use it as `qstack.qml` if it's installed?

**Comment by briling** [comment]
I also can't use it if qstack isn't installed but is just in the path (which is probably how one should not use it but...)

**Comment by liam-o-marsh** [comment]
yeah I can fix both of those

**Comment by briling** [comment]
Thanks!
I'm probably dumb but can't make it work (I can import `qstack.qml` but nothing further)

**Comment by liam-o-marsh** [comment]
I admittedly didn't test it.
I can't even `import qstack.qml`, only `from qstack import qml` or `import qstack; print(qstack.qml)`  (fixing this on my side for now)
I have no idea how to use b2r2, but I managed to run `qstack.qml.slatm.get_slatm_for_dataset()`

**Comment by briling** [comment]
thanks a lot! one still can't do `from qstack_qml.slatm import get_slatm_rxn` etc but I guess it's fine

**Comment by liam-o-marsh** [comment]
yeah, for this you need to have pip actually install qstack_qml (as one of qstack[all]'s dependencies)
the reason why qstack.qml works even when you didn't is because qstack/qml.py has a trick to modify PYTHONPATH, import, restore PYTHONPATH.

**Comment by briling** [comment]
oh sorry I can't spell. what I meant is this:

```>>> import qstack
>>> qstack.__path__
['/home/xe/SOFT/miniconda3/envs/qstack-yuri/lib/python3.10/site-packages/qstack']
>>> from qstack_qml.slatm import get_slatm_rxn
>>> from qstack.qml.slatm import get_slatm_rxn
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'qstack.qml.slatm'; 'qstack.qml' is not a package
>>> 
```

**Comment by briling** [comment]
While the following works and is pretty convenient so all good
```>>> from qstack.qml import slatm 
>>> slatm.get_slatm_rxn
<function get_slatm_rxn at 0x7f2fefd591b0>
>>> 
```

**Comment by liam-o-marsh** [comment]
(also unrelated but I finished the local kernel wrapper thing. Do you want me to put it on another branch?)

**Comment by briling** [comment]
> Do you want me to put it on another branch?
Thanks! yes please

Do you think we can merge (squash and merge?) this one?

**Comment by liam-o-marsh** [comment]
Yes, I think so.
(unless you absolutely want some sort of mechanism to ensure some coherence between environment.yml, requirements.txt and pyproject.toml, but I would have to find or implement this first)

**Comment by briling** [comment]
It's fine! It's not something that changes every day and we can do it manually
 
Do we need to look at/enable these alerts? https://github.com/lcmd-epfl/Q-stack/network/dependencies

File: qstack/regression/kernel_utils.py (hunk 1 above)
Comment (briling) [reply qstack/regression/kernel_utils.py:208]: #66

**Comment by liam-o-marsh** [comment]
don't think so, unless there's like a security issue somehow

**Pull request merged**